### ISSUE TITLE
Is it possible to set the Docker image's home directory to somewhere other than the mount point

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/ESCOMP-Containers#8]
@briandobbins - I'm not sure if what I'm asking for here is possible, but I thought I'd ask.... This came up for me in using the escomp/base image to build sphinx-based documentation, but I'm thinking that this is a more general issue.

Let's say I run the docker image using `-v ${HOME}/ctsm-repos:/home/user`, which I think is basically what you suggested that I do. Then the ${HOME}/ctsm-repos directory on my local system gets used as the home directory in Docker-land. This means that various hidden files and directories (`.whatever`) get created there, and these persist even after exiting from the Docker image. I found this confusing when I was examining directory contents later ("Where did these directories come from???" Since I had just recently been using Docker, I realized what must have happened, but if it had been a few days or weeks, I likely would have had no idea where those extra files and directories came from, and whether they were important.) In addition, it seems like this could lead to configuration settings made in one docker run in a given directory affecting the next docker run in that directory... which feels weird to me.

So what I'm wondering is whether it's possible to set up these Docker containers so that the host's mount point appears in Docker-land in some directory other than the home directory. I'm picturing something like mapping `${HOME}/ctsm-repos` in the host to `/home/user/workdir` in Docker-land. Then (in my mind) dot files that get written to your home directory would get written to `/home/user` in Docker-land, which is a directory that isn't mapped to anything on the host system, and so these dot files would not end up anywhere on the host's file system (which feels to me like more desirable behavior, though as you know, I'm very new to Docker, so I don't know what's typical).

A nice side-benefit of this is that you could mount your entire `${HOME}` directory into Docker without worrying about your host's dot files messing up Docker's operation, since the host's `${HOME}` would now appear as a subdirectory of the `${HOME}` directory in Docker-land.

Again, I'm not sure if what I'm asking for is possible....

**Comment by billsacks** [comment]
I went ahead and tried this out for the sphinx documentation build workflow:

```
docker run -i -t --rm -v ${HOME}:/home/user/mounted_home escomp/base
```

This seemed to work smoothly. I think I'm going to use this as the recommended workflow for the sphinx documentation build, unless you see any problems with it. Up to you whether you want to recommend this more generally.

**Comment by briandobbins** [comment]
@billsacks 

As you've discovered, this is definitely doable, and no, there's no problems that I see _for this kind of workflow_.  Basically, it comes down to whether persistence / customization is a 'feature' or a 'bug' in a particular use case.  S

My thinking is that for a very specific workflow like yours, where customization of commands and the addition of things like aliases or scripts is unnecessary, mounting into a 'home' directory that persists between runs of the container (or works in other ones) is more of a 'bug'.  You have these exact steps, people run them, and it works.

However, for more _interactive_ use, like in the CESM container, people may wish to customize their ~/.bashrc file, or add some common scripts they run for analysis, or something like that.  Here, we _don't_ want them to lose any changes made to their home directory.  One way to do that is to force them to save over their container every time with changes, but the easier and more flexible way is to just have them treat the mount directory as specifically for the Docker instance.  

This approach lets me switch between Docker instances (eg, CESM 2.1, CESM 2.2 and a CentOS 8 one) and keep all my scripts and customized environment.  In fact, you can even run multiple containers using the _same_ mount directory, so you only need to customize things once, and share files across them that way. 

In other words, it just comes down to the workflow.  I _think_ for interactive use, a customizable, persistent directory is a better default, and for a predefined workflow, I can see picking a mount point that won't put 'bookkeeping' hidden files there to avoid confusion.  And, naturally, experienced users can pick and choose, or use both.

(I'll leave this up as an open issue for a bit in case others have thoughts.)

**Comment by billsacks** [comment]
Thanks a lot for your thoughts @briandobbins . Everything you say makes sense, and I agree with your rationale. In particular, I agree with your assessment that these two workflows are different, and so the ideal docker setup will likely be different as well.

As far as I'm concerned, you can close this issue whenever you want.
